covalent/cloud_resource_manager/core.py

```diff
@@ -389,7 +397,24 @@ def _get_tf_statefile_path(self) -> str:
 
         """
         # Saving in a directory which doesn't get deleted on purge
-        return str(Path(self.executor_tf_path) / "terraform.tfstate")
+        return str(Path(self.executor_tfstate_path) / "terraform.tfstate")
+
+    def _get_tf_state_path(self, parent_path) -> str:
+        """
+        Get the terraform state file path
+
+        Args:
+            None
+
+        Returns:
+            Path to terraform state file
+
+        """
+        # Saving in a directory which doesn't get deleted on purge
+        state_path = os.path.join(parent_path, self.executor_name)
+        if not os.path.exists(state_path):
+            os.makedirs(state_path)
```

Comment:
Same comment as above. One minor thing is that if possible let's try to use `pathlib`'s `Path` instead of `os.path` to join/checking existence/checking whether its a directory/etc.